Syntax.md

```diff
@@ -0,0 +1,184 @@
+
+```html
+<p><span>Custom element definitions</span> are awesome.</p>
+```
+
+If you need to use a `<span>` or `<code>` element without it being a cross-reference, then you can use an empty `data-x=""` attribute:
+
+```html
+<p>In the following document, the element definition for <code
+data-x="">img-viewer</code> is loaded asynchronously:</p>
+```
+
+### Cross-specification cross references
+
+Wattsi does not support any database of definitions across all specifications in the ecosystem (like Bikeshed does). To work around this, the HTML Standard has a <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#dependencies">Dependencies</a> section where we create "proxy" definitions.
+
+These proxy definitions use a new attribute, `data-x-href=""`, which indicates the external URL where the definition was originally located. For example:
+
+```html
+<p>The following terms are defined in the WHATWG Infra standard: <ref spec=INFRA></p>
+
+<ul class="brief">
+ <li>The general iteration terms <dfn data-x-href="https://infra.spec.whatwg.org/#iteration-while">while</dfn>,
+         <dfn data-x-href="https://infra.spec.whatwg.org/#iteration-continue">continue</dfn>, and
+         <dfn data-x-href="https://infra.spec.whatwg.org/#iteration-break">break</dfn>.</li>
+ <li><dfn data-x-href="https://infra.spec.whatwg.org/#code-point">code point</dfn> and its synonym
+     <dfn data-x-href="https://infra.spec.whatwg.org/#code-point">character</dfn></li>
+ <li><dfn data-x-href="https://infra.spec.whatwg.org/#surrogate">surrogate</dfn></li>
+ <li><dfn data-x-href="https://infra.spec.whatwg.org/#scalar-value">scalar value</dfn></li>
+ <li><dfn data-x-href="https://infra.spec.whatwg.org/#noncharacter">noncharacter</dfn></li>
+ ...
+</ul>
+```
+
+Note how the above examples all use the default canonical definition text for the `<dfn>`. Sometimes we want to use a variant, as in the following example where the term is too common to be used by itself:
+
+```html
+<li>The <dfn data-x-href="https://infra.spec.whatwg.org/#list">list</dfn> data structure and the associated definitions for
+        <dfn data-x="list append" data-x-href="https://infra.spec.whatwg.org/#list-append">append</dfn>,
+        <dfn data-x="list replace" data-x-href="https://infra.spec.whatwg.org/#list-remove">replace</dfn>,
+        ...</li>
+```
+
+We can then use these "proxy definitions" as normal:
+
+```html
+<p>Each <code>DOMStringList</code> object has an associated <span>list</span>.</p>
+
+<p><span data-x="list append">Append</span> <var>serializedEntry</var> to
+<var>serialized</var>.[[SetData]].</p>
+```
+
+### Exporting definitions
+
+Definitions that are meant to be used by other specifications need to be exported using the [Bikeshed definition data model](https://tabatkins.github.io/bikeshed/#dfn-contract). In practice, this means adding `data-export=""` to the `<dfn>`, and if appropriate, `data-dfn-type=""`, `data-lt=""`, or `data-dfn-for=""` attributes. Some examples:
```

Comment:
Maybe link to the definitions directly?
https://tabatkins.github.io/bikeshed/#local-lt
https://tabatkins.github.io/bikeshed/#dfn-types
https://tabatkins.github.io/bikeshed/#dfn-for